SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -224,33 +230,48 @@ private Task ProcessPendingErrorsAsync()
                         {
                             AppCenterLog.Warn(LogTag, $"Failed to delete error log file {file.Name}.", ex);
                         }
+                        continue;
+                    }
+                    var report = BuildErrorReport(log);
+                    if (report == null)
+                    {
+                        AppCenterLog.Error(LogTag, $"Error parsing error log. Deleting invalid file: {file.Name}");
+                        RemoveAllStoredErrorLogFiles(log.Id);
                     }
                     else
                     {
-                        _unprocessedManagedErrorLogs.Add(log.Id, log);
+                        if (lastSessionErrorReport == null || lastSessionErrorReport.AppErrorTime < report.AppErrorTime)
+                        {
+                            lastSessionErrorReport = report;
+                        }
+                        if (ShouldProcessErrorReport?.Invoke(report) ?? true)
+                        {
+                            _unprocessedManagedErrorLogs.Add(log.Id, log);
+                        }
+                        else
+                        {
+                            AppCenterLog.Debug(LogTag, $"ShouldProcessErrorReport returned false, clean up and ignore log: {log.Id}");
+                            RemoveAllStoredErrorLogFiles(log.Id);
+                        }
                     }
                 }
-                ErrorReport lastSessionErrorReport = null;
-                if (lastSessionErrorLog != null)
-                {
-                    AppCenterLog.Debug(LogTag, "Setting last session error report to an actual report.");
-
-                    // TODO: Build error report from cache.
-                    lastSessionErrorReport = new ErrorReport(lastSessionErrorLog, null);
-                }
-                else
-                {
-                    AppCenterLog.Debug(LogTag, "Setting last session error report to null.");
-                }
                 _lastSessionErrorReportTaskSource.SetResult(lastSessionErrorReport);
                 await SendCrashReportsOrAwaitUserConfirmationAsync().ConfigureAwait(false);
             });
         }
 
+        private void RemoveAllStoredErrorLogFiles(Guid errorId)
+        {
+            // ReSharper disable once InconsistentlySynchronizedField this is a concurrent dictionary.
```

Comment:
Seems I'm told to delete comments about ReSharper as we progress - Is this helpful?

Reply:
We get warnings otherwise.